ui/src/components/SidePanel/SidePanel.tsx

```diff
@@ -90,11 +91,19 @@ const SidePanelComponent: FC<SidePanelProps> = ({
   isOverlay,
   isSplit = false,
   loading = false,
-  hasError,
+  hasError = false,
   className,
   width,
   pinned,
 }) => {
+  let content: ReactNode = null;
+  if (loading) {
+    content = <Loader />;
+  } else if (hasError) {
+    content = Label.ERROR_LOADING;
+  } else {
+    content = children;
+  }
```

Comment:
> suggestion(non-blocking): `useMemo` is useful here

Thanks for the suggestion. `useMemo` could be helpful, but I think that would be a micro optimisation at best at this stage. Almost all rerenders to the panel will be caused by `children` changing (i.e. when updating forms) and `children` would need to be a dependency on the `useMemo` which would cause the memo to recalculate on each child update.

My suggestion would be to leave this as is unless we hit a performance issue and then we look at what specific optimisations need to be made.

Reply:
That's fine. Yes my main concern was unnecessary rerenderings due to the change of the reference
But your analysis convinced me. Thank you :)